### PR TITLE
Created custom telemetry packet report

### PR DESCRIPTION
**Pull request by kb1lqc**
Modeled off of the PositionReport class, I created a TelemetryReport and
used similar serialization to create a working packet that was uploaded to
aprs.fi for testing.

This is my first iteration of working code that I confirmed would be accepted:

image

The string above has a blank comment at the end.

This code also does no input checking. One could break it by entering analog values > 255 or strings for their values. A larger or smaller array of digital values entered would also cause some issues.

I'd like to get your take on the code and I'd preferably ad some documentation and a Docstring unless you think otherwise. Thoughts?

### CODE REVIEW

**Comment by rossengeorgiev** [comment]
Oh cool. So here are my thoughts:

* Input validation. I've just pushed changes to `PositionReport` that validate `latitude` and `longitude`. See https://github.com/rossengeorgiev/aprs-python/commit/7d5dafa649625b9d0d1628514fb9fa2e273c1a04

* I don't like each value being a separate parameter, instead of a `list`. Currently the module will parse comment embedded telemetry and output a `list`. Besides consistency, it will simplify input validation

https://github.com/rossengeorgiev/aprs-python/blob/7d5dafa649625b9d0d1628514fb9fa2e273c1a04/aprslib/parsing/telemetry.py#L28-L33

File: aprslib/packets/telemetry.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            str(self.analog3).zfill(3),
+            str(self.analog4).zfill(3),
+            str(self.analog5).zfill(3),
+            str(self.digitalvalue),
+            self.comment,
+        ]
+        tmpbody = ",".join(body)
+
+        # remove static but erroneous comma between T# and sequenceno
+        return tmpbody[:2] + tmpbody[3:]
```
Comment (rossengeorgiev) [inline aprslib/packets/telemetry.py]: Could be made into  `return 'T#' + ','.join(body)`

File: aprslib/packets/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class TelemetryReport(APRSPacket):
+    format = 'raw'
+    sequenceno = 0
+    analog1 = 0
+    analog2 = 0
+    analog3 = 0
+    analog4 = 0
+    analog5 = 0
+    digitalvalue = ['0']*8
```
Comment (rossengeorgiev) [inline aprslib/packets/telemetry.py]: Just keep the value as a `int` as is easier to manipulate. Then for serialization do `'{:b}'.format(self.digitalvalue)`

**Comment by kb1lqc** [comment]
OK @rossengeorgiev I updated per your comments:

- Values are now stored as lists in a dictionary similar to how you showed in the comment telemetry
- I return with joined body tacked onto a packet type indicator string to avoid a string slice to remove an extra comma.

I did not yet perform any validation. I understand attribute setter with a single variable but don't know how to get it to work with a dictionary of lists. Thoughts?

**Comment by kb1lqc** [comment]
What do you think about validation @rossengeorgiev?

**Comment by rossengeorgiev** [comment]
I see why you did them that way, but comment telemetry is an addon to position reports. This is a telemetry report, and should have the values as properties like before. I would still keep the vals as `list` and modify/validate them through a setter function. Their names don't need to necessary line up with the names from the `dict`. If you want I can give it a go?

**Comment by kb1lqc** [comment]
@rossengeorgiev if you want feel free to give it a go from my work if you'd like. Glad to help where I can.

**Pull request merged**